### PR TITLE
Setup automated testing on Jenkins CI

### PR DESCRIPTION
**Pull request by dalg24**
@mkstoyanov this is a work in progress.  Do not merge yet.

### CODE REVIEW

**Comment by dalg24** [comment]
@mkstoyanov this ready for your review

File: docker/jenkins/build.sh [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+set -e
+
+[ -d build ] && rm -rf build
+mkdir build && cd build
+cmake \
+  -D CMAKE_CXX_FLAGS="-Wall -Wextra" \
+  -D CMAKE_CXX_COMPILER=g++ \
+  -D PYTHON_EXECUTABLE=/usr/bin/python3 \
+  -D Tasmanian_ENABLE_MATLAB=OFF \
```
Comment (mkstoyanov) [inline docker/jenkins/build.sh]: I fixed that, MATLAB is disabled by default and this line is not needed any more.

File: docker/jenkins/build.sh [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  ..
+make -j${NPROC}
+ctest -j${NPROC} --no-compress-output -T Test
```
Comment (mkstoyanov) [inline docker/jenkins/build.sh]: Do not use "ctest -j***". By default, Tasmanian uses OpenMP and -j can lead to more threads than cores and end up with slower runtime. Also, there are only 4 ctests (if you enable Fortran), multi-threading will probably not help here.

File: docker/jenkins/build.sh (hunk 1 above)
Comment (dalg24) [reply docker/jenkins/build.sh]: Will do.

File: docker/jenkins/build.sh (hunk 2 above)
Comment (dalg24) [reply docker/jenkins/build.sh]: With CMake you can set properties on a test and specify how many process slots are required.

**Pull request merged**